experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/NodeManagerCallbackHandler.scala

```diff
@@ -39,4 +40,15 @@ class NodeManagerCallbackHandler(am: ActorRef) extends NMClientAsync.CallbackHan
   def onStopContainerError(containerId: ContainerId, throwable: Throwable) {
     LOG.error(s"Container exception : $containerId", throwable)
   }
-}
\ No newline at end of file
+}
+
+trait NodeManagerCallbackHandlerFactory {
```

Comment:
Added to [PR](https://github.com/TomaszT/gearpump/pull/6) on Tomek's branch
Not all factories have been eliminated but most
```
class AmActor(appConfig: AppConfig, yarnConf: YarnConfiguration,
              rmCallbackHandlerActorProps: RMCallbackHandlerActorProps,
              rmClientActorProps: RMClientActorProps,
              createNMClient: (NodeManagerCallbackHandler, YarnConfiguration) => NMClientAsync,
              createNodeManagerCallbackHandler: (ActorRef) => NodeManagerCallbackHandler,
              containerLaunchContextFactory: ContainerLaunchContextFactory) extends Actor {
```